api/python/provisioner/commands/set_swupgrade_repo.py

```diff
@@ -150,6 +157,40 @@ def _single_repo_validation(self, release, repo_name):
                     )
                 )
 
+    @staticmethod
+    def _validate_python_index(index_path: Path):
+        """
+        Perform the dynamic validation for SW upgrade Python index by
+        the given index path `index_path`
+
+        Parameters
+        ----------
+        index_path: Path
+            Path to the SW upgrade Python index
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        SWUpdateRepoSourceError
+            If Python index validation fails
+
+        """
+        logger.debug("Start Python index validation")
+        test_package_name = next(index_path.iterdir())
+        cmd = (f"pip3 download {test_package_name} --dest=/tmp/ "
+               f"--find-links file://{index_path.absolute()}")
```

Comment:
would it use the local index only on that way?

Reply:
Yes, it would. I tested different options, cases, and variants. This is one of the suitable options for the case of dynamic validation.
As an example, you can see, that during execution pip looks at packages only in the path specified by `--find-links` option. It is not true for `--extra-index-url`:
```
pip download gami --dest=~/tmp/ --find-links  file:///mnt/python_deps/
Looking in links: file:///mnt/python_deps/
ERROR: Could not find a version that satisfies the requirement gami
ERROR: No matching distribution found for gami
```

Reply:
I changed `--find-links` to `--index-url` because it is most reliable option. It works with any pip configuration when you have `index-url` , `trusted-host`, `extra-index-url` options already configured in `pip.conf`.